Samples build - duplication symbols error

**Describe the bug**
While compiling SDK samples (cmake, x86_64 GNU/Linux, gcc 10), linker reports the following duplicated symbols error:
```
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x0): multiple definition of `iot_sample_hub_hostname_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x0): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x80): multiple definition of `iot_sample_provisioning_id_scope_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x80): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0xa0): multiple definition of `iot_sample_hub_device_id_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0xa0): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0xe0): multiple definition of `iot_sample_provisioning_registration_id_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0xe0): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x1e0): multiple definition of `iot_sample_hub_sas_key_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x1e0): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x260): multiple definition of `iot_sample_provisioning_sas_key_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x260): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x2e0): multiple definition of `iot_sample_x509_cert_pem_file_path_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x2e0): first defined here
/usr/bin/ld: libaz_iot_sample_common.a(iot_sample_common.c.o):(.bss+0x3e0): multiple definition of `iot_sample_x509_trust_pem_file_path_buffer'; CMakeFiles/paho_iot_hub_c2d_sample.dir/paho_iot_hub_c2d_sample.c.o:(.bss+0x3e0): first defined here
collect2: error: ld returned 1 exit status
gmake[2]: *** [sdk/samples/iot/CMakeFiles/paho_iot_hub_c2d_sample.dir/build.make:114: sdk/samples/iot/paho_iot_hub_c2d_sample] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:400: sdk/samples/iot/CMakeFiles/paho_iot_hub_c2d_sample.dir/all] Error 2
```

The root-case is definition of the above symbols (arrays) in `sdk/samples/iot/iot_sample_common.h` and including the header in `iot_sample_common.c` (which is a common object for the samples build) + any other SDK samples executables. Looking at the sources I don't see a reason to expose these arrays in the header. Shouldn't they be moved into `iot_sample_common.c` as static symbols?

- OS: Linux
- Version of the Library used: 1.2.0

**Information Checklist**
Please make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report

- [X] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

--- a/sdk/samples/iot/iot_sample_common.c
+++ b/sdk/samples/iot/iot_sample_common.c
@@ -43,6 +43,18 @@
     }                                           \
   } while (0)
 
+static char iot_sample_hub_hostname_buffer[128];
+static char iot_sample_provisioning_id_scope_buffer[16];
+
+static char iot_sample_hub_device_id_buffer[64];
+static char iot_sample_provisioning_registration_id_buffer[256];
+
+static char iot_sample_hub_sas_key_buffer[128];
+static char iot_sample_provisioning_sas_key_buffer[128];
+
+static char iot_sample_x509_cert_pem_file_path_buffer[256];
+static char iot_sample_x509_trust_pem_file_path_buffer[256];
+
 //
 // MQTT endpoints
 //

--- a/sdk/samples/iot/iot_sample_common.h
+++ b/sdk/samples/iot/iot_sample_common.h
@@ -115,18 +115,6 @@ bool get_az_span(az_span* out_span, char const* const error_message, ...);
 // This is usually not needed on Linux or Mac but needs to be set on Windows.
 #define IOT_SAMPLE_ENV_DEVICE_X509_TRUST_PEM_FILE_PATH "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE_PATH"
 
-char iot_sample_hub_hostname_buffer[128];
-char iot_sample_provisioning_id_scope_buffer[16];
-
-char iot_sample_hub_device_id_buffer[64];
-char iot_sample_provisioning_registration_id_buffer[256];
-
-char iot_sample_hub_sas_key_buffer[128];
-char iot_sample_provisioning_sas_key_buffer[128];
-
-char iot_sample_x509_cert_pem_file_path_buffer[256];
-char iot_sample_x509_trust_pem_file_path_buffer[256];
-
 typedef struct
 {
   az_span hub_device_id;

